lib/command/src/Obelisk/Command/Run.hs

```diff
@@ -105,38 +107,50 @@ runWatch = do
   pkgs <- getLocalPkgs
   withGhciScript pkgs $ \dotGhciPath -> runGhcid dotGhciPath Nothing
 
+data GuessPackageFileError = GuessPackageFileError_Ambiguous [FilePath] | GuessPackageFileError_NotFound
+  deriving (Eq, Ord, Show)
+instance Exception GuessPackageFileError
+
+-- | Given a directory, try to guess what the appropriate @.cabal@ or @package.yaml@ file is for the package.
+guessCabalPackageFile
+  :: MonadObelisk m
+  => FilePath -- ^ Directory to search for cabal package
+  -> m (Either GuessPackageFileError (Either FilePath FilePath))
+  -- ^ 'Right' 'Left' for hpack package, 'Right' 'Right' for cabal package; 'FilePath' is relative to given directory.
+guessCabalPackageFile dir = do
+  candidates <- liftIO $
+        filterM doesFileExist
+    =<< filter (liftA2 (||) (== Hpack.packageConfig) (".cabal" `L.isSuffixOf`))
+    <$> listDirectory dir
+  pure $ case L.partition (== Hpack.packageConfig) candidates of
+    ([hpack], _) -> Right $ Left hpack
```

Comment:
This also catches `([hpack], [cabal])`. I'm not too familiar with `hpack`, I guess we can assume that to be the generated cabal from hpack *if* the name matches?
I don't think this should catch `([hpack], [cabal1,cabal2])` at any rate.

Reply:
I'm going with "If you have a package.yaml file, I don't care what your cabal situation is because it should be generated anyway"